src/Sylius/Behat/Context/Domain/ManagingProductsContext.php

```diff
@@ -66,31 +66,26 @@ public function __construct(
 
     /**
      * @When /^I delete the ("[^"]+" variant of product "[^"]+")$/
-     */
-    public function iDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
-    {
-        $this->sharedStorage->set('product_variant_id', $productVariant->getId());
-        $this->productVariantRepository->remove($productVariant);
-    }
-
-    /**
      * @When /^I try to delete the ("[^"]+" variant of product "[^"]+")$/
      */
-    public function iTryToDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
+    public function iDeleteTheVariantOfProduct(ProductVariantInterface $productVariant)
```

Comment:
There is a reason for keeping `I delete` and `I try to delete` methods separate. For domain implementation we mute the exception in step `I try to delete` so we can assert it occurred, but for step `I delete` we shouldn't do that and let it be thrown. This change should be reverted.